launchable/commands/test_path_writer.py

```diff
@@ -3,17 +3,18 @@
 
 import click
 
+from ..app import Application
 from ..testpath import TestPath
 
 
 class TestPathWriter(object):
     base_path: Optional[str] = None
 
-    def __init__(self, dry_run=False):
-        self._formatter = TestPathWriter.default_formatter
+    def __init__(self, app: Application):
+        self._formatter = TestPathWriter.default_formatter  # type: Callable[[TestPath], str]
         self._separator = "\n"
-        self._same_bin_formatter = None
-        self.dry_run = dry_run
+        self._same_bin_formatter = None  # type: Optional[Callable[[str], Dict[str, str]]]
```

Comment:
When I made my change, type checker started complaining, which was strange. More context at https://github.com/launchableinc/cli/pull/716/commits/9435ec06207af3f7c4dcddafbdfdce237d6cc9cf if you care.